pkg/virt-api/rest/interfacehotplug_test.go

```diff
@@ -348,4 +348,94 @@ var _ = Describe("Interface Hotplug Subresource", func() {
 			).To(BeEmpty())
 		})
 	})
+
+	Context("Remove Interface Subresource api", func() {
+		const (
+			ifaceToHotUnplug    = "pluggediface1"
+			existingNetworkName = ifaceToHotUnplug
+		)
+
+		BeforeEach(func() {
+			request.PathParameters()["name"] = "testvm"
+			request.PathParameters()["namespace"] = "default"
+		})
+
+		AfterEach(func() {
+			restoreKubeVirtClusterConfig()
+		})
+
+		createVMI := func(removeOpts *v1.RemoveInterfaceOptions) *v1.VirtualMachineInstance {
+			mutateIfaceRequest(request, removeOpts)
+			vmi := api.NewMinimalVMI(request.PathParameter("name"))
+			vmi.Namespace = "default"
+			vmi.Status.Phase = v1.Running
+			vmi.Spec.Domain.Devices.Interfaces = append(vmi.Spec.Domain.Devices.Interfaces, v1.Interface{
+				Name: existingNetworkName,
+			})
+			vmi.Spec.Networks = append(vmi.Spec.Networks, v1.Network{
+				Name: existingNetworkName,
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{
+						NetworkName: existingNetworkName,
+					},
+				},
+			})
+			return vmi
+		}
+
+		successfulMockScenarioForVMI := func(removeOpts *v1.RemoveInterfaceOptions) {
+			vmi := createVMI(removeOpts)
+
+			vmiClient.EXPECT().Get(context.Background(), vmi.Name, &k8smetav1.GetOptions{}).Return(vmi, nil)
+			vmiClient.EXPECT().Patch(context.Background(), vmi.Name, types.JSONPatchType, gomock.Any(), &k8smetav1.PatchOptions{}).Return(vmi, nil)
+
+			if removeOpts != nil {
+				app.VMIRemoveInterfaceRequestHandler(request, response)
+			}
+		}
+
+		It("Should succeed a remove interface request on VMI", func() {
+			enableFeatureGate(virtconfig.HotplugNetworkIfacesGate)
+			removeOpts := &v1.RemoveInterfaceOptions{Name: ifaceToHotUnplug}
+			successfulMockScenarioForVMI(removeOpts)
+
+			Expect(response.StatusCode()).To(Equal(http.StatusAccepted))
+		})
+
+		It("Should generate expected vmi patch when the add interface request features all required data", func() {
+			vmi := api.NewMinimalVMI(request.PathParameter("name"))
+			vmi.Namespace = "default"
+			vmi.Status.Phase = v1.Running
+			vmi.Spec.Domain.Devices.Interfaces = append(vmi.Spec.Domain.Devices.Interfaces, v1.Interface{
+				Name: existingNetworkName,
+			})
+			vmi.Spec.Networks = append(vmi.Spec.Networks, v1.Network{
+				Name: existingNetworkName,
+			})
+
+			Expect(
+				generateVMIInterfaceRequestPatch(
+					vmi,
+					&v1.VirtualMachineInterfaceRequest{
+						RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: ifaceToHotUnplug},
+					},
+				),
+			).To(
+				Equal(
+					fmt.Sprintf(`[{ "op": "test", "path": "/spec/networks", "value": [{"name":%[1]q}]}, { "op": "test", "path": "/spec/domain/devices/interfaces", "value": [{"name":%[1]q}]}, { "op": "add", "path": "/spec/networks", "value": [{"name":%[1]q}]}, { "op": "add", "path": "/spec/domain/devices/interfaces", "value": [{"name":%[1]q,"state":"absent"}]}]`, existingNetworkName),
```

Comment:
>There is a comment about this one at the production code, it can be discussed there.
Ref: https://github.com/kubevirt/kubevirt/commit/f24df1e255783c0d35580f28e8968e2b67a8621b?diff=split&w=0#r1204303286

I cant find the right thread, can you paste the updated link?

> q adds the quotes and the [1] just uses the same (first) argument (existingNetworkName) .

Thanks, I totally missed that one.

> I cannot add newlines, that will cause it not to equal. I could normalize it, but this seems to be an effort required for the existing hotplug part as well, so we can do it as a follow up.

You can use the `MatchJSON` matcher, it compare JSONs in a way that spaces and tabs are ignored, see those tests for example: https://github.com/kubevirt/kubevirt/blob/4e6d36b1842edada8bbce6b4cf160b565cb3fe2b/pkg/virt-controller/watch/vmi_test.go#L3248-L3275
Still a nit though..

Reply:
> I cant find the right thread, can you paste the updated link?

https://github.com/kubevirt/kubevirt/pull/9779/files#r1204303286

> You can use the MatchJSON matcher

Nice, Done.